Cluster load fixture sometimes fails with `AttributeError: 'NoneType' object has no attribute 'get'`

See https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-deploy-ocs-cluster-prod/91/testReport/junit/tests.manage.z_cluster.cluster_expansion.test_add_capacity/TestAddCapacity/test_add_capacity/

http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/j006ai3c33-t1/j006ai3c33-t1_20210304T102734/logs/

```
Error Message
test teardown failure
Stacktrace
def finalizer():
        """
        Stop the thread that executed watch_load()
        """
        config.RUN["load_status"] = "finished"
        if thread:
            thread.join()
        if cluster_load_error:
>           raise cluster_load_error

tests/conftest.py:1227: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/conftest.py:1212: in cluster_load
    cl_load_obj = ClusterLoad()
ocs_ci/ocs/cluster_load.py:81: in __init__
    self.pvc_size = int(get_osd_pods_memory_sum() * 0.5)
ocs_ci/ocs/cluster.py:1189: in get_osd_pods_memory_sum
    osd_pod_mem_size_str = osd_pods[0].get_memory().get("osd")
ocs_ci/ocs/resources/pod.py:457: in get_memory
    for container in containers
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

.0 = <list_iterator object at 0x7fc9036cd860>

    container_names_and_memory = {
        container.get("name"): container.get("resources")
        .get("limits")
        .get("memory")
>       for container in containers
    }
E   AttributeError: 'NoneType' object has no attribute 'get'

ocs_ci/ocs/resources/pod.py:457: AttributeError
```

--- a/ocs_ci/ocs/cluster.py
+++ b/ocs_ci/ocs/cluster.py
@@ -1186,7 +1186,7 @@ def get_osd_pods_memory_sum():
     """
     osd_pods = pod.get_osd_pods()
     num_of_osd_pods = len(osd_pods)
-    osd_pod_mem_size_str = osd_pods[0].get_memory().get("osd")
+    osd_pod_mem_size_str = osd_pods[0].get_memory(container_name=constants.OSD)
     osd_pod_mem_size = convert_device_size(
         unformatted_size=osd_pod_mem_size_str, units_to_covert_to="GB"
     )

--- a/ocs_ci/ocs/resources/pod.py
+++ b/ocs_ci/ocs/resources/pod.py
@@ -440,23 +440,27 @@ class Pod(OCS):
         cmd = f'ssh -i {authkey} -o "StrictHostKeyChecking no" {user}@{server} {cmd}'
         self.exec_cmd_on_pod(cmd, out_yaml_format=False)
 
-    def get_memory(self):
+    def get_memory(self, container_name):
         """
         Get the pod memory size
 
+        Args:
+            container_name (str): The name of the container to look for
+
         Returns:
-            dict: The names of the pod's containers (str) as keys and their memory
-                size (str) as values
+            str: The container memory size (e.g. '5Gi')
 
         """
-        containers = self.pod_data.get("spec").get("containers")
-        container_names_and_memory = {
-            container.get("name"): container.get("resources")
-            .get("limits")
-            .get("memory")
-            for container in containers
-        }
-        return container_names_and_memory
+        pod_containers = self.pod_data.get("spec").get("containers")
+        matched_containers = [
+            c for c in pod_containers if c.get("name") == container_name
+        ]
+        if len(matched_containers) > 1:
+            logger.error(
+                f"Multiple containers, of the same name, were found: {[c.get('name') for c in matched_containers]}"
+            )
+        container = matched_containers[0]
+        return container.get("resources").get("limits").get("memory")
 
     def get_node(self):
         """

